updatem/orchestration/update_orchestrator_feedback.go

```diff
@@ -0,0 +1,419 @@
+		orchestrator.handleDomainIdentificationFailed(domain, message)
+	case types.StatusCompleted:
+		orchestrator.handleDomainCompletedEvent(domain)
+	case types.StatusIncomplete:
+		orchestrator.handleDomainIncomplete(domain, message)
+	case types.BaselineStatusDownloading:
+		orchestrator.handleDomainDownloading(domain)
+	case types.BaselineStatusDownloadSuccess:
+		orchestrator.handleDomainDownloadSuccess(domain)
+	case types.BaselineStatusDownloadFailure:
+		orchestrator.handleDomainDownloadFailure(domain, message)
+	case types.BaselineStatusUpdating:
+		orchestrator.handleDomainUpdating(domain)
+	case types.BaselineStatusUpdateSuccess:
+		orchestrator.handleDomainUpdateSuccess(domain)
+	case types.BaselineStatusUpdateFailure:
+		orchestrator.handleDomainUpdateFailure(domain, message)
+	case types.BaselineStatusActivating:
+		orchestrator.handleDomainActivating(domain)
+	case types.BaselineStatusActivationSuccess:
+		orchestrator.handleDomainActivationSuccess(domain)
+	case types.BaselineStatusActivationFailure:
+		orchestrator.handleDomainActivationFailure(domain, message)
+	case types.BaselineStatusCleanup:
+		orchestrator.handleDomainCleanup(domain)
+	case types.BaselineStatusCleanupSuccess:
+		orchestrator.handleDomainCleanupSuccess(domain)
+	case types.BaselineStatusCleanupFailure:
+		orchestrator.handleDomainCleanupFailure(domain, message)
+	}
+}
+
+func (orchestrator *updateOrchestrator) validateActivity(domain, activityID, baseline string, status types.StatusType) bool {
+	if orchestrator.operation == nil {
+		logger.Warn("received desired state feedback event for domain [%s], but there is no active update operation", domain)
+		return false
+	}
+	if orchestrator.operation.activityID != activityID {
+		logger.Warn("activity id mismatch for received desired state feedback event for domain [%s]  - expecting %s, received %s", domain, orchestrator.operation.activityID, activityID)
+		return false
+	}
+	if _, ok := orchestrator.operation.domains[domain]; !ok {
+		logger.Warn("received desired state feedback event for unexpected domain [%s]", domain)
+		return false
+	}
+	if status != types.StatusIdentified && status != types.StatusIdentificationFailed &&
+		status != types.StatusCompleted && status != types.StatusIncomplete &&
+		status != types.BaselineStatusDownloading && status != types.BaselineStatusDownloadSuccess && status != types.BaselineStatusDownloadFailure &&
+		status != types.BaselineStatusUpdating && status != types.BaselineStatusUpdateSuccess && status != types.BaselineStatusUpdateFailure &&
+		status != types.BaselineStatusActivating && status != types.BaselineStatusActivationSuccess && status != types.BaselineStatusActivationFailure &&
+		status != types.BaselineStatusCleanup && status != types.BaselineStatusCleanupSuccess && status != types.BaselineStatusCleanupFailure &&
+		status != types.BaselineStatusRollback && status != types.BaselineStatusRollbackSuccess && status != types.BaselineStatusRollbackFailure {
```

Comment:
can we use some array with with possible statuses that we can check, or a map or basically whatever other solution you can think of beside this `if` monster?

Reply:
Also, if we receive an event with unsupported status is it still valid? I guess the function should return false in that case?